examples/vision/barlow_twins.py

```diff
@@ -0,0 +1,1046 @@
+
+        Arguments:
+            x: A tf.Tensor representing the image.
+
+        Returns:
+            Returns a flipped version of the image 50% of the time
+              and the original image 50% of the time.
+        """
+
+        if self.random_execute(0.5):
+            x = tf.image.random_flip_left_right(x)
+        return x
+
+
+class RandomResizedCrop(Augmentation):
+    """RandomResizedCrop class.
+
+    RandomResizedCrop class. Randomly crop an image to a random size,
+    then resize the image back to the original size.
+
+    Methods:
+        __call__: Method that does random resize crop to the image.
+    """
+
+    def call(self, x: tf.Tensor, image_size: int) -> tf.Tensor:
+        """__call__ function.
+
+        Does random resize crop by randomly cropping an image to a random
+        size 75% - 100% the size of the image. Then resizes it.
+
+        Arguments:
+            x: A tf.Tensor representing the image.
+
+        Returns:
+            Returns a randomly cropped image.
+        """
+
+        rand_size = tf.random.uniform(
+            shape=[],
+            minval=int(0.75 * image_size),
+            maxval=1 * image_size,
+            dtype=tf.int32,
+        )
+
+        crop = tf.image.random_crop(x, (rand_size, rand_size, 3))
+        crop_resize = tf.image.resize(crop, (image_size, image_size))
+        return crop_resize
+
+
+class RandomSolarize(Augmentation):
+    """RandomSolarize class.
+
+    RandomSolarize class. Randomly solarizes an image.
+    Solarization is when pixels accidentally flip to an inverted state.
+
+    Methods:
+        __call__: Method that does random solarization 20% of the time.
+    """
+
+    @tf.function
+    def call(self, x: tf.Tensor) -> tf.Tensor:
+        """__call__ function.
+
+        Randomly solarizes the image.
+
+        Arguments:
+            x: A tf.Tensor representing the image.
+
+        Returns:
+            Returns a solarized version of the image 20% of the time
+              and the original image 80% of the time.
+        """
+
+        if self.random_execute(0.2):
+            # flips abnormally low pixels to abnormally high pixels
+            x = tf.where(x < 10, x, 255 - x)
+        return x
+
+
+class RandomBlur(Augmentation):
+    """RandomBlur class.
+
+    RandomBlur class. Randomly blurs an image.
+
+    Methods:
+        __call__: Method that does random blur 20% of the time.
+    """
+
+    @tf.function
+    def call(self, x: tf.Tensor) -> tf.Tensor:
+        """__call__ function.
+
+        Randomly solarizes the image.
+
+        Arguments:
+            x: A tf.Tensor representing the image.
+
+        Returns:
+            Returns a blurred version of the image 20% of the time
+              and the original image 80% of the time.
+        """
+
+        if self.random_execute(0.2):
+            s = np.random.random()
+            return tfa.image.gaussian_filter2d(image=x, sigma=s)
+        return x
+
+
+class RandomAugmentor:
```

Comment:
How about if we use keras layer instead. 

```python

class RandomAugmentor(keras.layers.Layer):
     def __init__():
          init all_augmentation_layers.

     def call(x):
          calling sequentially.
```

Reply:
Maybe keras model would make more sense because the layers are being chained?

Reply:
Converted it into a keras.Model class